modules/decklinksrc/decklinksrc.py

```diff
@@ -28,28 +31,40 @@ class DeckLinkSRC:
         Displays live video feed of DeckLink stream
     """
 
-    def __init__(self):
+    def __init__(self, videoPipeline):
         """
         Initializes DeckLink video stream
         """
         self.__currentFrame = None
-        self.capture = cv2.VideoCapture('decklinkvideosrc mode=7 connection=0 ! videoconvert ! appsink') #Starts capture on initialization of object
+        self.capture = cv2.VideoCapture(
+            'decklinkvideosrc mode=7 connection=0 ! videoconvert ! appsink')  # Starts capture on initialization of object
+        self.videoPipeline = videoPipeline
+        mainProcess = mp.Process(target=self._mainProcess_, daemon=True)
+        mainProcess.start()
+
+    def _mainProcess_(self):
+        print("decklink started")
+        while True:
+            newFrame = self.grab()
+            if self.videoPipeline.empty() or newFrame != self.videoPipeline.get():
+                self.videoPipeline.put(newFrame)
+            sleep(0.1)
 
     def stop(self):
         """
         Logic for stopping video feed by releasing capture and destroying any windows open.
         """
         self.capture.release()
         cv2.destroyAllWindows()
-    
 
-    def grab(self, frame):
+    def grab(self):  # Logic for capturing single frame from deckLink
         """
         Logic for grabbing single frame from DeckLink
         """
-        cv2.imshow('Grabbedframe', self.__currentFrame)
-    
-    def get_frame(self):
+        self.__currentFrame = self.capture.read()
+        return self.get_frame()
+
+    def get_frame(self):  # Logic for returning the current frame as a numpy array
```

Comment:
Same as above comment

Reply:
Ok imo the get_frame function makes no sense at all, so I'll just take it out, none of the class attributes are private so no need for a getter.